source/extensions/filters/http/abac/abac_filter.cc

```diff
@@ -0,0 +1,171 @@
+#include "extensions/filters/http/abac/abac_filter.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ABACFilter {
+
+using CelValue = google::api::expr::runtime::CelValue;
+using CelMap = google::api::expr::runtime::CelMap;
+using CelList = google::api::expr::runtime::CelList;
+
+namespace {
+
+constexpr absl::string_view kMetadata = "metadata";
+constexpr absl::string_view kConnection = "connection";
+constexpr absl::string_view kRequestedServerName = "requested_server_name";
+constexpr absl::string_view kLocalAddress = "local_address";
+constexpr absl::string_view kRemoteAddress = "remote_address";
+constexpr absl::string_view kCertificatePresented = "certificate_presented";
+constexpr absl::string_view kSubjectLocalCertificate = "subject_local_certificate";
+constexpr absl::string_view kIssuerPeerCertificate = "issuer_peer_certificate";
+constexpr absl::string_view kSubjectPeerCertificate = "subject_peer_certificate";
+constexpr absl::string_view kTlsVersion = "tls_version";
+constexpr absl::string_view kRequest = "request";
+constexpr absl::string_view kPath = "path";
+constexpr absl::string_view kHost = "host";
+constexpr absl::string_view kScheme = "scheme";
+constexpr absl::string_view kMethod = "method";
+constexpr absl::string_view kReferer = "referer";
+constexpr absl::string_view kHeaders = "headers";
+constexpr absl::string_view kTime = "time";
+
+class ConnectionWrapper : public CelMap {
+public:
+  ConnectionWrapper(const StreamInfo::StreamInfo& info) : info_(info) {}
+
+  absl::optional<CelValue> operator[](CelValue key) const override {
+    if (!key.IsString()) {
+      return {};
+    }
+    auto value = key.StringOrDie().value();
+    if (value == kRequestedServerName) {
```

Comment:
@jplevyak We could do something like this in WASM environment interface. It'd be even better if we shared this piece, but we just need to align on the mapping of names really.